### PR TITLE
Clean Field component

### PR DESCRIPTION
**Pull request by mickaelzhang**
## What does this do?

No UI changes but cleaned up some of the props that were doing nothing

### CODE REVIEW

File: src/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -21,7 +21,6 @@ export type DropdownItem = {
 }
 
 export interface Props extends CommonFieldTypes {
-  ref?: RefObject<HTMLSelectElement>
```
Comment (mickaelzhang) [inline src/Dropdown/Dropdown.tsx:24]: This is useless

File: src/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -74,9 +72,6 @@ export const Dropdown = forwardRef(function Dropdown(
       onSelect(list[0].value)
     }
 
-    // update 'key' props in order to deselect the active option if a new list passed
-    list.length > 1 ? setKey(`${list[0].value}-${list.length}`) : ''
```
Comment (mickaelzhang) [inline src/Dropdown/Dropdown.tsx:78]: This was a fix from 3 years ago but couldn't reproduce
This seems more like an hack than a real fix so I was thinking of removing it for now

File: src/Field/Field.tsx [hunk 3]
```diff
@@ -72,21 +59,7 @@ export const Field = ({
...
-        outlined={outlined}
-        error={error}
-        key={dropdownKey ?? null}
-      >
-        {children}
-        {showCaret && (
-          <Caret outlined={outlined}>
-            <Icon render="caret" color="subtext" size={24} />
-          </Caret>
-        )}
```
Comment (mickaelzhang) [inline src/Field/Field.tsx:87]: Caret has been moved to Dropdown since it's only used there

File: src/Field/Field.tsx [hunk 4]
```diff
@@ -72,21 +59,7 @@ export const Field = ({
         </>
       )}
 
-      <Content
-        fullHeight={fullHeight}
-        value={value}
-        outlined={outlined}
-        error={error}
-        key={dropdownKey ?? null}
```
Comment (mickaelzhang) [inline src/Field/Field.tsx:80]: Checked and none of the props were doing anything

File: src/Fieldset/Fieldset.tsx [hunk 5]
```diff
@@ -1,11 +1,11 @@
-import React from 'react'
+import React, { ReactNode } from 'react'
 
 import { MarginProps } from '../utils/space'
 import { Box } from '../Box'
 import { Text } from '../Text'
 
 type FieldSetProps = {
-  children: React.ReactChild
+  children: ReactNode
```
Comment (mickaelzhang) [inline src/Fieldset/Fieldset.tsx:8]: Update to be able to be more flexible in what is passed as children

**Review by Jmuskett — APPROVED** [review]

**Pull request merged**